arrow-libs/core/arrow-eval/src/commonMain/kotlin/arrow/eval/Eval.kt

```diff
@@ -0,0 +1,579 @@
+package arrow.eval
+
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.getOrElse
+import arrow.typeclasses.Monoid
+import arrow.typeclasses.MonoidDeprecation
+import kotlin.js.JsName
+import kotlin.jvm.JvmStatic
+
+/**
+ * Eval is a monad which controls evaluation of a value or a computation that produces a value.
+ *
+ * Three basic evaluation strategies:
+ *
+ *  - Now:    evaluated immediately
+ *  - Later:  evaluated once when value is needed
+ *  - Always: evaluated every time value is needed
+ *
+ * The Later and Always are both lazy strategies while Now is eager.
+ * Later and Always are distinguished from each other only by
+ * memoization: once evaluated Later will save the value to be returned
+ * immediately if it is needed again. Always will run its computation
+ * every time.
+ *
+ * methods, which use an internal trampoline to avoid stack overflows.
+ * Computation done within .map and .flatMap is always done lazily,
+ * even when applied to a Now instance.
```

Comment:
This text seems to be incomplete

Reply:
The file is just restored from the original version :/